event-driven-architecture/src/main/java/com/iluwatar/eda/App.java

```diff
@@ -53,12 +53,12 @@
   public static void main(String[] args) {
 
     EventDispatcher dispatcher = new EventDispatcher();
-    dispatcher.registerChannel(UserCreatedEvent.class, new UserCreatedEventHandler());
```

Comment:
Here you could actually have something like this:
```
dispatcher.registerChannel(UserCreatedEvent.class, new UserCreatedEventHandler());
dispatcher.registerChannel(UserUpdatedEvent.class, new UserUpdatedEventHandler());
```

The updated dispatcher will not allow this.